packages/build/src/plugins/plugin_conditions.ts

```diff
@@ -81,7 +83,7 @@ const siteDependencyTest = async function ({
 
   try {
     // if this is a range we need to get the exact version
-    const packageJsonPath = await resolvePath(`${dependencyName}/package.json`, buildDir)
+    const packageJsonPath = await resolvePath(`${dependencyName}/package.json`, join(buildDir, packagePath || ''))
```

Comment:
Very similar to fix in https://github.com/netlify/build/pull/6105, where other part of the system was not taking `packagePath` into consideration

Do note that if dependency is in `node_modules` in root of monorepo - we still will be able to resolve, because `require.resolve` is used and not direct FS check. This is also verified with added test case for hoisted `node_modules` still passing